File: Java/src/SelectionSort.java
public class SelectionSort {

    static void sort(int[] arr){
        /*
         * This is the function that will sort the array using selection sort. Selection sort works by finding the
         * smallest element from the unsorted side and placing it in the correct place by swapping.
         */
        for (int i = 0;i<arr.length;i++){
            int minidx = i;
            for (int j = i;j<arr.length;j++){
                if (arr[j] < arr[minidx]){
                    minidx = j;
                }
            }
            int tmp = arr[i];
            arr[i] = arr[minidx];
            arr[minidx] = tmp;
        }
    }

    public static void main(String[] args) {
        SelectionSort selectionSort = new SelectionSort();
    }

}

File: C++/selectionSort.cpp
//
// Created by harry on 20/03/2023.
//

void selectionSort(int arr[], int n) {
    for (int i = 0; i < n; i++) {
        int minIdx = i;
        for (int j = i+1; j<n;j++){
            if (arr[minIdx] > arr[j]){
                minIdx = j;
            }
        }
        int tmp = arr[minIdx];
        arr[minIdx] = arr[i];
        arr[i] = tmp;
    }
}

